Sources/HummingbirdWSCore/WebSocketHandler.swift

```diff
@@ -193,12 +177,34 @@ package actor WebSocketHandler {
                 try? await self.close(code: .normalClosure)
             }
         }
-        return switch self.closeState {
+        return switch self.stateMachine.state {
         case .closed(let code): code
         default: nil
         }
     }
 
+    func runAutoPingLoop() async throws {
+        let period = self.stateMachine.pingTimePeriod
+        try await Task.sleep(for: period)
+        loop: while true {
+            switch self.stateMachine.sendPing() {
+            case .sendPing(let buffer):
+                try await self.write(frame: .init(fin: true, opcode: .ping, data: buffer))
+
+            case .wait(let time):
+                try await Task.sleep(for: time)
+
+            case .closeConnection(let errorCode):
+                try await self.sendClose(code: errorCode, reason: "Ping timeout")
+                try await self.channel.close(mode: .input)
+                break loop
+
+            case .stop:
+                break loop
```

Comment:
We can just return instead of breaing the loop right?

Reply:
That way we don't even need a label

Reply:
done